ScoutSuite/providers/azure/authentication_strategy.py

```diff
@@ -1,87 +1,49 @@
 import json
 import logging
-import requests
 from getpass import getpass
-from datetime import datetime, timedelta
-
-from azure.common.credentials import ServicePrincipalCredentials, UserPassCredentials, get_azure_cli_credentials
-from msrestazure.azure_active_directory import MSIAuthentication
-from ScoutSuite.core.console import print_info, print_debug, print_exception
-from msrestazure.azure_active_directory import AADTokenCredentials
-import adal
+import requests
+from ScoutSuite.core.console import print_exception
 
+from azure.identity import UsernamePasswordCredential,AzureCliCredential, ClientSecretCredential, \
+    ManagedIdentityCredential, InteractiveBrowserCredential
 from ScoutSuite.providers.base.authentication_strategy import AuthenticationStrategy, AuthenticationException
 
-
-AUTHORITY_HOST_URI = 'https://login.microsoftonline.com'
+AUTHORITY_HOST_URI = 'https://login.microsoftonline.com/'
 AZURE_CLI_CLIENT_ID = "04b07795-8ddb-461a-bbee-02f9e1bf7b46"
 
 
 class AzureCredentials:
 
     def __init__(self,
-                 arm_credentials, aad_graph_credentials,
+                 identity_credentials,
                  tenant_id=None, default_subscription_id=None,
                  context=None):
 
-        self.arm_credentials = arm_credentials  # Azure Resource Manager API credentials
-        self.aad_graph_credentials = aad_graph_credentials  # Azure AD Graph API credentials
+        self.identity_credentials = identity_credentials  # Azure Resource Manager API credentials
         self.tenant_id = tenant_id
         self.default_subscription_id = default_subscription_id
         self.context = context
 
     def get_tenant_id(self):
         if self.tenant_id:
             return self.tenant_id
-        elif 'tenant_id' in self.aad_graph_credentials.token:
-            return self.aad_graph_credentials.token['tenant_id']
+        elif hasattr(self.identity_credentials,'tenant_id'):
+            return self.identity_credentials['tenant_id']
+
         else:
-            # This is a last resort, e.g. for MSI authentication
+            # Additional request for CLI & MSI authentication
             try:
-                h = {'Authorization': 'Bearer {}'.format(self.arm_credentials.token['access_token'])}
+                access_token = self.identity_credentials.get_token("https://management.core.windows.net/")
+                h = {'Authorization': 'Bearer {}'.format(access_token.token)}
```

Comment:
Where possible, use f-strings. Appreciate `format()` is used in much of the codebase though.